openmdao/approximation_schemes/complex_step.py

```diff
@@ -0,0 +1,213 @@
+"""Finite difference derivative approximations."""
+from __future__ import division, print_function
+
+import numpy as np
+from collections import namedtuple
+from itertools import groupby
+from six.moves import range
+
+from openmdao.approximation_schemes.approximation_scheme import ApproximationScheme
+from openmdao.utils.name_maps import abs_key2rel_key
+
+
+DEFAULT_CS_OPTIONS = {
+    'step': 1e-6,
```

Comment:
The default step size should be much smaller, say 1e-15, because 1e-15 will always be as accurate as or more accurate than 1e-6